docker/scripts/entrypoint.sh

```diff
@@ -20,92 +30,51 @@ NH_SECRET_PHRASE_PATH=${NH_SECRET_PHRASE_PATH:-"/data/secret_phrase.dat"}
 echo "NH_SECRET_PHRASE_PATH=${NH_SECRET_PHRASE_PATH}"
 
 # Node configurations (env->arg)
-NH_CONF_NAME=${NH_CONF_NAME:-"MyNode"}
-NH_CONF_BASE_PATH=${NH_CONF_BASE_PATH:-"/data/node"}
-NH_CONF_CHAIN=${NH_CONF_CHAIN:-"test"}
-NH_CONF_VALIDATOR=${NH_CONF_VALIDATOR:-}
-NH_CONF_NODE_KEY_FILE=${NH_CONF_NODE_KEY_FILE:-}
-NH_CONF_BOOTNODES=${NH_CONF_BOOTNODES:-}
-NH_CONF_RPC_CORS=${NH_CONF_RPC_CORS:-}
-NH_CONF_RPC_EXTERNAL=${NH_CONF_RPC_EXTERNAL:-}
-NH_CONF_RPC_METHODS=${NH_CONF_RPC_METHODS:-}
-NH_CONF_PRUNING=${NH_CONF_PRUNING:-}
-NH_PROMETHEUS_EXTERNAL=${NH_PROMETHEUS_EXTERNAL:-}
-
-for var_name in NH_CONF_NAME NH_CONF_BASE_PATH NH_CONF_CHAIN NH_CONF_VALIDATOR NH_CONF_NODE_KEY_FILE NH_CONF_BOOTNODES NH_CONF_RPC_CORS NH_CONF_RPC_EXTERNAL NH_CONF_RPC_METHODS NH_CONF_PRUNING NH_PROMETHEUS_EXTERNAL; do
-  # Get the value of the variable
-  var_value="${!var_name}"
-
-  # Check if the variable is defined
-  if [ -n "${var_value}" ]; then
-    echo "${var_name}=${var_value}"
-  else
-    echo "${var_name} is empty"
+prefix="NH_CONF_"
+conf_args=""
+echo "Node configuration:"
+while IFS='=' read -r -d '' var_name var_value; do
+  if [[ "$var_name" == NH_CONF* ]]; then
+    arg_name=$(get_arg_name_from_env_name "${var_name}" "${prefix}")
+    arg_value=""
+    if [ "$var_value" != "true" ]; then
+      arg_value=" ${var_value}"
+    fi
+    arg="${arg_name}${arg_value}"
+    conf_args+="${arg} "
+    echo "  ${var_name}=${var_value} -> $arg"
   fi
-done
+done < <(env -0)
 
+# Keys handling
 if [ -f "${NH_SECRET_PHRASE_PATH}" ]; then
+  path=""
+  if [ -z "${NH_CONF_BASE_PATH+}" ]; then
+    path=$(get_arg_name_from_env_name NH_CONF_BASE_PATH "${prefix}") $NH_CONF_BASE_PATH
+  fi
+  chain=""
```

Comment:
I am not totally familiar of all the use cases this script would be used in, but could there be instances where the lack of explicitly setting `NH_CONF_BASE_PATH` and `NH_CONF_CHAIN` variables would be possible? In this case, and only if this may break things due to unintended behavior, I would add a check that stops the script (or at least warns the user) if either `path` and `chain` local variables result to be empty even after reading them from `env`.

Reply:
The cases of `NH_CONF_BASE_PATH` and/or `NH_CONF_CHAIN` missing should be fine; btw, in a production environment they would be explicitly defined as environment variables.

Reply:
Thanks for the clarification!